packages/ai-jsx/src/index.ts

```diff
@@ -180,17 +181,22 @@ export function createContext<T>(defaultValue: T): Context<T> {
   return ctx;
 }
 
-const indirectWeakMap = new WeakMap<IndirectNode, Node>();
 export function isIndirectNode(value: unknown): value is IndirectNode {
-  return indirectWeakMap.has(value as IndirectNode);
+  if (value !== null && typeof value === 'object') {
+    return indirectNodeSymbol in value;
+  }
+  return false;
 }
 
 export function getIndirectNode(value: IndirectNode): Node {
```

Comment:
This name is a little funny, given that it says `getIndirectNode` but returns `Node`.